### DISCUSSION TITLE
How to format figure from external file?

### DISCUSSION THREAD

**Discussion by pjotrhey** [quarto-dev/quarto-cli#10853] [Q&A]
### Description

Hi all,

I absolutely love Quarto so far, but I have one main issue that I cannot get solved and that is how to format a figure from an external file (either .qmd or .ipynb). As an example, this is my main file:

```
---
title: "Quarto doc w/ External Notebooks"
format: 
  pdf:
    fig-format: pdf
  html:
    fig-format: svg
editor: source
---

## Figure - internal
```{python}
#| echo: false
#| label: fig-internal
#| fig-cap: "Caption."

import numpy as np
import matplotlib.pyplot as plt

x = np.linspace(0,1,100)
y = x**2
plt.plot(x, y)
``[and the third chunk delimiter, but then the layout messes up]

plt.xlabel('X-Axis Label')
plt.ylabel('Y-Axis Label')
plt.title('Sample Plot')
plt.show()

## Figure from external Quarto file
{{< embed example-plot.qmd#fig-external-qmd >}}

## Figure from external Jupyter notebook
{{< embed example-plot.ipynb#fig-external-ipynb >}}
```

And then this is example-plot.qmd:

```
---
title: External notebook - qmd
authors:
  - name: Author name
format: 
  pdf:
    fig-format: pdf
  html:
    fig-format: svg
---

```{python}
#| label: fig-external-qmd
#| fig-cap: "Caption."
 
import numpy as np
import matplotlib.pyplot as plt

x = np.linspace(0,1,100)
y = x**2
plt.plot(x, y)

plt.xlabel('X-Axis Label')
plt.ylabel('Y-Axis Label')
plt.title('Sample Plot')
plt.show()
```

The jupyter notebook file is basically the same (it can be found here: https://github.com/pjotrhey/quarto-fig-external). 

Now if I format my main file to either PDF or HTML, the output of the figure is a .png instead of either PDF or SVG. What am I doing wrong here? 

I'm using Quarto 1.5.57.

Thanks in advance!

**Comment by cscheid** [comment]
If you're using a figure from an external notebook, that notebook controls the format (since it's already been rendered) and you end up with whatever format the external notebook uses.

**Reply by pjotrhey** [reply]
Thanks for your answer!

So I have some follow-up questions:
1) Is there no option to set the format based on whether my main script is going to output HTML (then I want .svg) or PDF (then I want .pdf)? I prefer vector based images rather than raster (png) that will work for both output.
2) The alternative then I think is to render at a high dpi. I use the yaml below and this works while rendering the .qmd, but not for the .ipynb. What am I doing wrong?
```
fig-format: png
fig-dpi: 600 (or something)
```

Thanks again!
